### PR TITLE
Ticket list 

### PR DESCRIPTION
**Pull request by sarykemmoun**

### CODE REVIEW

File: js/components/Tickets/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        const modified = new Date(lastModifiedDate);
+        const now = new Date();
+        const diffMS = now - modified;
+        const diffD = diffMS / 1000 / 60 / 60 / 24;
+        if (diffD < 15) {
+            return true;
+        }
+        return false;
+    }
```
Comment (raphaelboukara) [inline js/components/Tickets/index.js]:
```js
isActivable({ status, lastModifiedDate }) {
        const modified = new Date(lastModifiedDate),
            now = new Date(),
            diffMS = now - modified,
            diffD = diffMS / 1000 / 60 / 60 / 24;

        return status !== 'Closed' || diffD < 15
    }
```
don't afraid about performance issue, there is no pblem here

File: js/redux/actions/TicketsActions.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export const getTickets = () => {
+    return (dispatch, getState) => {
+        const { refreshingList, pageIndex, ticketsList, totalItemCount } = getState().tickets;
+        if (!refreshingList && (ticketsList.length < totalItemCount || ticketsList.length === 0)) {
+            return dispatch({
+                type: GET_MY_TICKETS,
+                payload: {
+                    promise: axios.get(`${SERVER_URL}/api/ticketing/ticketuser/beneficiary/${'user_pfsa@demo.com'}/${pageIndex}`)
+                }
+            });
```
Comment (raphaelboukara) [inline js/redux/actions/TicketsActions.js]: else?

File: js/components/Tickets/index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    style={FABStyle}
+                    position='bottomRight'
+                    onPress={() => console.log('sndms')}
+                >
+                    <Icon name='add' />
+                </Fab>
+            </View>
+        );
+    }
+}
```
Comment (raphaelboukara) [inline js/components/Tickets/index.js]: We need to speak about this component. This component is too much, the purpose is (after refactor) to have something like this:
```js
function Tickets() {
    return (
        <Scene>
            <Search/>
            <Tickets/>
            <Fab/>
        </Scene>
    )
}
```
We will speak about that next week

File: js/components/Tickets/TicketItem.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+class TicketItem extends Component {
+    render() {
+        const { status, title, lastModifiedDate, appointmentDate, isRead } = this.props.ticket;
+        const date = new Date(appointmentDate);
+        const { spaceBetween, flexRow, calendar, calendarIcon, RDVText, notRead, clock, clockIcon, lastModifiedText } = styles;
+        return (
+            <TicketLayout
+                onPress={() => { }}
+                isRead={isRead}
+                label={{ color: status === 'Closed' ? green : orange, char: status.charAt(0).toUpperCase() }}
```
[the remark this answers is not in the document]
Comment (sarahmarciano) [reply js/components/Tickets/TicketItem.js]: We are not sure you are right here,
the TicketItem component is responsible for rendering the ticket, I'm not sure that the ticketIlist needs to do this logic, The ticket list doesn't need to know about this UI level of status etc...
all what the ticket list has to do is:
`return <TicketItem {...ticket.item} />;`
and then the ticketItem will do his logic for rendering the ticket 
What do you think?

File: js/redux/actions/TicketsActions.js (hunk 2 above)
Comment (sarahmarciano) [reply js/redux/actions/TicketsActions.js]: There is nothing to do in the else.
Should I do 
`else {
      //do nothing
}`
is it necessary?

File: js/components/Tickets/index.js (hunk 3 above)
Comment (philjoseph) [reply js/components/Tickets/index.js]: I hope to be after this re-factoring before Sukkot @raphaelboukara 
Also for now I did not succeed to have @sarahmarciano or @sarykemmoun available for Wednesday that is "Erev hag".

File: js/components/Tickets/TicketItem.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (raphaelboukara) [reply js/components/Tickets/TicketItem.js]: When you mean `ticket list` you mean `TicketLayout` ?

[the remark this answers is not in the document]
Comment (raphaelboukara) [reply js/components/Tickets/TicketItem.js]: `The ticket list doesn't need to know about this UI level of status` I 100% agree, but I didn't say the ticket list need to handle the logic...

File: js/redux/actions/TicketsActions.js (hunk 2 above)
Comment (raphaelboukara) [reply js/redux/actions/TicketsActions.js]: not sur

**Pull request merged**